drivers/misc/mcux_flexio/mcux_flexio.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, STRIM, ALC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_MCUX_FLEXIO_H_
+#define ZEPHYR_DRIVERS_MCUX_FLEXIO_H_
+
+#include <zephyr/device.h>
+
+
+struct mcux_flexio_child_res {
+	uint8_t *shifter_index;
+	uint8_t shifter_count;
+	uint8_t *timer_index;
+	uint8_t timer_count;
+};
+
+typedef int (*mcux_flexio_child_isr_t)(const struct device *dev);
```

Comment:
A few comments on this callback signature:
- shouldn't we use the `dev` parameter to contain the FlexIO controller device itself? This would mirror the way other callbacks within Zephyr are designed
- Again following the pattern of other callbacks in Zephyr, maybe we should add a parameter `user_data`, which is defined within the `mcux_flexio_child` structure?